presto-spi/src/main/java/com/facebook/presto/spi/PageBuilder.java

```diff
@@ -49,7 +57,7 @@ public PageBuilder(List<? extends Type> types)
 
         blockBuilders = new BlockBuilder[types.size()];
         for (int i = 0; i < blockBuilders.length; i++) {
-            blockBuilders[i] = types.get(i).createBlockBuilder(pageBuilderStatus.createBlockBuilderStatus(), 1, pageBuilderStatus.getMaxBlockSizeInBytes());
+            blockBuilders[i] = types.get(i).createBlockBuilder(pageBuilderStatus.createBlockBuilderStatus(), expectedEntries, pageBuilderStatus.getMaxBlockSizeInBytes());
```

Comment:
The last argument should be `pageBuilderStatus.getMaxBlockSizeInBytes() / expectedEntries`